### PR TITLE
EHN: Gat score optional epochs

### PR DESCRIPTION
**Pull request by kingjr**
@cmoutard @dengemann @agramfort 

This is a small EHN that allows using gat.score() without passing the epochs if and only if the user has already used gat.predict(). 

As the object already stores the epochs predictions (in gat.y_pred_) it already possess all the necessary information which can be useful when you want to happened to have already predicted your epochs and want to test several different scorers.

### CODE REVIEW

File: mne/decoding/tests/test_time_gen.py [hunk 1]
```diff
@@ -244,3 +244,11 @@ def test_generalization_across_time():
...
                         gat.fit(epochs, y=y_)
                         gat.score(epochs, y=y_)
+
+    # Test score without passing epochs
+    gat = GeneralizationAcrossTime()
+    with warnings.catch_warnings(record=True):
+        gat.fit(epochs)
+    assert_raises(RuntimeError, gat.score)
+    gat.predict(epochs)
+    gat.score()
```
Comment (agramfort) [inline mne/decoding/tests/test_time_gen.py]: don't refit for just testing this. It's a waste of test time

**Comment by agramfort** [comment]
travis is not happy

File: mne/decoding/tests/test_time_gen.py (hunk 1 above)
Comment (kingjr) [inline mne/decoding/tests/test_time_gen.py]: ok

**Comment by dengemann** [comment]
+1 for merge if @kingjr is happy.

**Comment by kingjr** [comment]
Happy I am

**Comment by dengemann** [comment]
Thanks @kingjr !

**Comment by agramfort** [comment]
>
> Thanks @kingjr <https://github.com/kingjr> !
>
+1

**Pull request merged**